torch/csrc/inductor/aoti_runtime/model.h

```diff
@@ -268,7 +273,37 @@ class AOTInductorModelBase {
           cudaMemcpyHostToDevice));
     }
     return internal_ptr;
-#else // !USE_CUDA
+#elif USE_MMAP_SELF
+    // get pointer to constant which is packed in model during compile time.
+    AOTI_RUNTIME_CHECK(!skip_copy, "pure cpu mode doesn't support skip copy");
+    if (!self_mmap) {
+      Dl_info dl_info;
+      // get pointer to constant which are appended to the binary
+      AOTI_RUNTIME_CHECK(
+          dladdr(__func__, &dl_info), "Can't find shared library name");
+      int fd = open(dl_info.dli_fname, O_RDONLY);
+      AOTI_RUNTIME_CHECK(fd >= 0, "Shared library file cannot be opened");
+      auto fsize = lseek(fd, 0, SEEK_END);
+      auto weights_size =
+          reinterpret_cast<const uint64_t*>(_binary_constants_bin_start)[0];
+      auto magic_number =
+          reinterpret_cast<const uint64_t*>(_binary_constants_bin_start)[1];
+      auto weights_offset = fsize - weights_size;
+      AOTI_RUNTIME_CHECK(
+          (weights_offset & 4095) == 0,
```

Comment:
I've updated it in both cases to 0x4000 as it will be compatible with both old a new and up to 16Kb of empty space is really peanuts compared to 2+Gb file